File: python/test/00-example.py
#!/bin/bash

##############################################################################
# BOOTSTRAP
#
# Use python3 or python, whichever is available.
# (See https://unix.stackexchange.com/questions/20880)
#
if "true" : '''\'
then
    exec $(command -v python3 || command -v python) "$0" "$@"
    exit 127
fi
'''


##############################################################################
# PYTHON CODE BEGINS HERE

import sys
import liteexpr


def main():
    symbols = liteexpr.SymbolTable({
        "grades" : {
            "alice" : "A",
            "bob"   : "B",
        }
    })

    liteexpr.eval("""
        PRINT("I have " + LEN(grades) + " students");
        PRINT("Alice's grade is " + grades.alice);
    """, symbols)


##############################################################################
# ENTRY POINT

if __name__ == "__main__":
    main()


# vim:filetype=python:

File: cpp/test/00-example.cpp
#include <string>
#include <iostream>
#include "liteexpr.h"

int main(int argc, const char* argv[]) {
    liteexpr::SYMBOLS symbols = liteexpr::make_symbols({
        { "grades", liteexpr::make_value({
            { "alice", liteexpr::make_value("A") },
            { "bob"  , liteexpr::make_value("B") },
        })},
    });

    try {
        liteexpr::eval(R"(
            PRINT("I have " + LEN(grades) + " students");
            PRINT("Alice's grade is " + grades.alice);
        )", symbols);
    }
    catch(liteexpr::Error e) {
        std::cerr << std::string(e) << std::endl;
        return 1;
    }

    return 0;
}
